src/process/id.rs

```diff
@@ -106,44 +106,52 @@ impl Gid {
 impl Pid {
     /// A `Pid` corresponding to the init process (pid 1).
     pub const INIT: Self = Self(
-        // SAFETY: The init process' pid is always valid.
+        // SAFETY: One is non-zero.
         unsafe { RawNonZeroPid::new_unchecked(1) },
     );
 
     /// Converts a `RawPid` into a `Pid`.
     ///
+    /// Returns `Some` for strictly positive `RawPid`s. Otherwise, returns `None`.
+    ///
     /// # Safety
     ///
-    /// `raw` must be the value of a valid Unix process ID, or zero.
+    /// This is always safe because a `Pid` is a number without any guarantees for the kernel.
+    /// Non-child `Pid`s are always racy for any syscalls, but can only cause logic errors. If you
+    /// want race-free access or control to non-child processes, please consider other mechanisms
+    /// like [pidfd] on Linux.
+    ///
+    /// [pidfd]: https://man7.org/linux/man-pages/man2/pidfd_open.2.html
     #[inline]
-    pub const unsafe fn from_raw(raw: RawPid) -> Option<Self> {
-        match RawNonZeroPid::new(raw) {
-            Some(pid) => Some(Self(pid)),
-            None => None,
+    pub const fn from_raw(raw: RawPid) -> Option<Self> {
+        if raw > 0 {
+            // SAFETY: raw > 0.
+            unsafe { Some(Self::from_raw_unchecked(raw)) }
+        } else {
+            None
         }
     }
 
-    /// Converts a known non-zero `RawPid` into a `Pid`.
+    /// Converts a known strictly positive `RawPid` into a `Pid`.
     ///
     /// # Safety
     ///
-    /// `raw` must be the value of a valid Unix process ID. It must not be
-    /// zero.
+    /// The caller must guarantee `raw` is strictly positive.
     #[inline]
-    pub const unsafe fn from_raw_nonzero(raw: RawNonZeroPid) -> Self {
-        Self(raw)
+    pub const unsafe fn from_raw_unchecked(raw: RawPid) -> Self {
+        // FIXME: `const_panic` is stabilized since Rust 1.57.
+        // debug_assert!(raw > 0);
```

Comment:
Ideally, we want to have this `debug_assert`. But it needs to bump MSRV.
It is not required to be correct, though.

Reply:
It's looking like we'll bump the MSRV at our next semver bump, so this should be ok. If you want, you can bump the MSRV in this PR to 1.63, so that you can re-enable this assert.

Reply:
> If you want, you can bump the MSRV in this PR to 1.63, so that you can re-enable this assert.

It'll collide with #651 I think?

Reply:
If you keep it as a separate commit, it should be pretty easy to drop when it's time to merge things.

1.63 is the version of Rust in the [next Debian stable release](https://packages.debian.org/bookworm/rustc), which is what rustix's MSRV policy is currently based on. It's expected to be released [next month](https://lists.debian.org/debian-devel-announce/2023/04/msg00007.html).

Reply:
> If you keep it as a separate commit, it should be pretty easy to drop when it's time to merge things.
> 
> 1.63 is the version of Rust in the [next Debian stable release](https://packages.debian.org/bookworm/rustc), which is what rustix's MSRV policy is currently based on. It's expected to be released [next month](https://lists.debian.org/debian-devel-announce/2023/04/msg00007.html).

I prefer waiting for another individual PR to do such a big MSRV bump, after the official Debian release. I'm afraid there are many unrelated code should also be cleaned up (both to make CI happy and avoid outdated comments).  Not a fan of mixing them here.